app/queries/subscriber_lists_for_finder_query.rb

```diff
@@ -0,0 +1,38 @@
+class SubscriberListsForFinderQuery
+  attr_reader :govuk_path
+
+  class NotAFinderError < StandardError; end
+
+  def initialize(govuk_path:)
+    @govuk_path = govuk_path
+  end
+
+  SELECT_FOR_TAGS_TEMPLATE = "SELECT
+    EXISTS(
+        SELECT
+            formats
+        FROM
+            (
+                SELECT
+                    json_array_elements_text(
+                        tags -> ? -> 'any'
+                    ) AS formats
+            ) AS allowed_formats
+        WHERE
+            formats = ?
+    )".freeze
+
+  def call
+    content_item = GdsApi.content_store.content_item(govuk_path).to_hash
+    raise NotAFinderError unless content_item["document_type"] == "finder"
+
+    lists = []
+    content_item["details"]["filter"].each_key do |filter_name|
+      Array(content_item["details"]["filter"][filter_name]).each do |filter_value|
+        lists += SubscriberList.where(SELECT_FOR_TAGS_TEMPLATE, filter_name, filter_value)
+      end
+    end
```

Comment:
Do we expect there to be anything other than `format` in the `filter` JSON? If we don't then this could be a lot simpler, e.g:
```
SubscriberList.where(SELECT_FOR_TAGS_TEMPLATE, "format",  content_item["details"]["filter"]["format"])
```

Reply:
No, we can definitely have other things than `format` (for instance search/news-and-communications has this:

```
filter: {
  content_purpose_subgroup: [
    "news",
    "speeches_and_statements"
  ]
},
```

Reply:
I think there's also some with `content_purpose_supergroup`.